docs/tools/hardhat/hardhat-zksync-verify-vyper.md

```diff
@@ -0,0 +1,167 @@
+
+:::
+
+## Setup
+
+The [@matterlabs/hardhat-zksync-verify-vyper](https://www.npmjs.com/package/@matterlabs/hardhat-zksync-verify-vyper) plugin is used to verify contracts on zkSync network.
+To use it, install plugin and then import `@matterlabs/hardhat-zksync-verify-vyper` in the `hardhat.config.ts` file.
+
+::: code-tabs
+
+@tab:active yarn
+
+```bash
+yarn add -D @matterlabs/hardhat-zksync-verify-vyper
+```
+
+@tab npm
+
+```bash
+npm i -D @matterlabs/hardhat-zksync-verify-vyper
+```
+
+:::
+
+### Configuration
+
+Import the plugin in the `hardhat.config.ts` file:
+
+```javascript
+import "@matterlabs/hardhat-zksync-verify-vyper";
+```
+
+Add the `verifyURL` property to the zkSync Era network in the `hardhat.config.ts` file as shown below:
+
+```typescript
+networks: {
+    goerli: {
+      url: "https://goerli.infura.io/v3/<API_KEY>" // The Ethereum Web3 RPC URL (optional).
+    },
+    zkTestnet: {
+      url: "https://testnet.era.zksync.dev", // The testnet RPC URL of zkSync Era network.
+      ethNetwork: "goerli", // The Ethereum Web3 RPC URL, or the identifier of the network (e.g. `mainnet` or `goerli`)
+      zksync: true,
+      // Verification endpoint for Goerli
+      verifyURL: 'https://zksync2-testnet-explorer.zksync.dev/contract_verification'
+    }
+},
+// defaultNetwork: "zkTestnet", // optional (if not set, use '--network zkTestnet')
+```
+
+Additional network properties:
+
+- `zkTestnet` is an arbitrary zkSync Era network name. You can select this as the default network using the `defaultNetwork` property.
+- `url` is a field with the URL of the zkSync Era node. This field is required for all zkSync networks used by this plugin.
+- `ethNetwork` is a field with the URL of the Ethereum node. You can also provide network name (e.g. `goerli`) as the value of this field. In this case, the plugin will either use the URL of the appropriate Ethereum network configuration (from the `networks` section), or the default `ethers` provider for the network if the configuration is not provided. This field is required for all zkSync networks used by this plugin.
+- `zksync` is a flag that indicates a zkSync Era network configuration. This field is set to `true` for all zkSync Era networks. Field value `true` is required for this plugin work. If field is missing or if values is set to `false` plugin will throw a error.
+- `verifyURL` is a field that points to the verification endpoint for the specific zkSync network. This parameter is optional, and its default value is the testnet verification url.
+  - Testnet: `https://zksync2-testnet-explorer.zksync.dev/contract_verification`
+  - Mainnet: `https://zksync2-mainnet-explorer.zksync.io/contract_verification`
+
+### Commands
+
+```sh
+yarn hardhat verify:vyper --network <network> <contract address>
```

Comment:
After installing with `yarn add -D ...` then running this command produces an error, any ideas? 

```bash
yarn hardhat verify:vyper
yarn run v1.22.19
$ /Users/dennis/Documents/Code/vyper-test/node_modules/.bin/hardhat verify:vyper
An unexpected error occurred:

Error: Cannot find module '@matterlabs/hardhat-zksync-verify'
Require stack:
- /Users/dennis/Documents/Code/vyper-test/hardhat.config.ts
- /Users/dennis/Documents/Code/vyper-test/node_modules/hardhat/internal/core/config/config-loading.js
- /Users/dennis/Documents/Code/vyper-test/node_modules/hardhat/internal/cli/cli.js
- /Users/dennis/Documents/Code/vyper-test/node_modules/hardhat/internal/cli/bootstrap.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1028:15)
    at Function.Module._resolveFilename.sharedData.moduleResolveFilenameHook.installedValue [as _resolveFilename] (/Users/dennis/Documents/Code/vyper-test/node_modules/@cspotcode/source-map-support/source-map-support.js:811:30)
    at Function.Module._load (node:internal/modules/cjs/loader:873:27)
    at Module.require (node:internal/modules/cjs/loader:1100:19)
    at require (node:internal/modules/cjs/helpers:108:18)
    at Object.<anonymous> (/Users/dennis/Documents/Code/vyper-test/hardhat.config.ts:7:1)
    at Module._compile (node:internal/modules/cjs/loader:1198:14)
    at Module.m._compile (/Users/dennis/Documents/Code/vyper-test/node_modules/ts-node/src/index.ts:1618:23)
    at Module._extensions..js (node:internal/modules/cjs/loader:1252:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/dennis/Documents/Code/vyper-test/node_modules/ts-node/src/index.ts:1621:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/dennis/Documents/Code/vyper-test/hardhat.config.ts',
    '/Users/dennis/Documents/Code/vyper-test/node_modules/hardhat/internal/core/config/config-loading.js',
    '/Users/dennis/Documents/Code/vyper-test/node_modules/hardhat/internal/cli/cli.js',
    '/Users/dennis/Documents/Code/vyper-test/node_modules/hardhat/internal/cli/bootstrap.js'
  ]
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Reply:
contents of package.json:

```json
{
  "name": "zksync-hardhat-vyper-template",
  "version": "1.0.0",
  "main": "index.js",
  "author": "Antonio <aug@matterlabs.dev>",
  "license": "MIT",
  "devDependencies": {
    "@matterlabs/hardhat-zksync-deploy": "^0.6.3",
    "@matterlabs/hardhat-zksync-verify-vyper": "^0.0.1-alpha.2",
    "@matterlabs/hardhat-zksync-vyper": "^0.2.0",
    "@nomiclabs/hardhat-etherscan": "^3.1.7",
    "@nomiclabs/hardhat-vyper": "^3.0.3",
    "@types/chai": "^4.3.4",
    "@types/mocha": "^10.0.1",
    "chai": "^4.3.7",
    "dotenv": "^16.3.1",
    "ethers": "^5.7.2",
    "hardhat": "^2.15.0",
    "mocha": "^10.2.0",
    "ts-node": "^10.9.1",
    "typescript": "^5.1.3",
    "zksync-web3": "^0.14.3"
  },
  "scripts": {
    "test": "NODE_ENV=test hardhat test --network zkSyncTestnet",
    "deploy": "yarn hardhat deploy-zksync --script deploy-greeter.ts",
    "greet": "yarn hardhat deploy-zksync --script use-greeter.ts"
  }
}
```